api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java

```diff
@@ -791,4 +791,26 @@ public void getCountOfDrugs_shouldReturnTheTotalNumberOfMatchingNumbers() {
 		assertEquals(new Integer(1),
 		    conceptService.getCountOfDrugs(phrase, conceptService.getConcept(conceptId), true, true, true));
 	}
+
+	/**
+	 * @see ConceptServiceImpl#getConceptNameTagByName(String)
+	 */
+	@Test
+	public void getConceptNameTagByName_shouldReturnNullIfNoConceptNameTagIsFound() {
+		assertNull(conceptService.getConceptNameTagByName(null));
+	}
+
+	/**
+	 * @see ConceptServiceImpl#getConceptNameTagByName(String) 
+	 */
+	@Test
+	public void getConceptNameTagByName_shouldReturnTheMatchingConceptNameTagObjectIfFound() {
+		ConceptNameTag c = new ConceptNameTag();
+		c.setTag("tag");
+		c.setDescription("some description");
+		assertNull(c.getId());
+		ConceptNameTag savedC = conceptService.saveConceptNameTag(c);
+		assertNotNull(savedC.getId());
+		assertEquals(savedC.getDescription(), conceptService.getConceptNameTagByName("tag").getDescription());
```

Comment:
Two different conceptNameTags can have the same description right? We can just assert that both of the objects are equal.

Reply:
Yes, thanks for pointing that out, I'll check for IDs.

Reply:
Also, you can write tests based on standardTestDataset.xml which is the test dataset. No need to create a new ConceptNameTag

Reply:
Thanks for suggestion, I'm working on that ;)